prom.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	log "github.com/Sirupsen/logrus"
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type promMessage struct {
+	registry *prom.Registry
+}
+
+type promIO struct {
+	scrapeSignalChan chan bool
+	messageChan      chan promMessage
+}
+
+func resetRegistry(registry *prom.Registry) {
+	prom.DefaultRegisterer = registry
```

Comment:
This seems odd. You're overwriting the global default registry.
Function name implies you are reseting a registry.
Fiddling with the global registry is almost certainly the wrong thing to do.